pkg/operator/controller/extension/controllerregistration/controllerregistration_test.go

```diff
@@ -0,0 +1,183 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package controllerregistration_test
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	gardencorev1 "github.com/gardener/gardener/pkg/apis/core/v1"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	"github.com/gardener/gardener/pkg/logger"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	. "github.com/gardener/gardener/pkg/operator/controller/extension/controllerregistration"
+)
+
+var _ = Describe("ControllerRegistration", func() {
+	var (
+		ctx                    context.Context
+		log                    logr.Logger
+		virtualClient          client.Client
+		controllerRegistration Interface
+
+		extensionName string
+		ociRef        string
+		extensionKind string
+
+		extension *operatorv1alpha1.Extension
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		log = logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, logzap.WriteTo(GinkgoWriter))
+		virtualClient = fakeclient.NewClientBuilder().WithScheme(operatorclient.VirtualScheme).Build()
+		controllerRegistration = New(&record.FakeRecorder{})
+
+		extensionName = "test-extension"
+		ociRef = "local-extension:v1.2.3"
```

Comment:
Use name of extension
```suggestion
		ociRef = "test-extension:v1.2.3"
```